### ISSUE TITLE
Windows 11, ColumnsReview ignores most column headers on the Desktop

### ISSUE THREAD

**Issue by amirsol81** [ABuffEr/columnsReview#34]
@ABuffEr Greetings. With Windows 11, ColumnsReview doesn't announce the Size, Item type, and Date modified headers for items on the Desktop. Worse, pressing Control+NVDA+2, 3 or 4 forces NVDA to incorrectly say, "No more columns available," though they are indeed available as JAWS can properly read/detect them with its Control+JAWS key+2, 3 and 4 key strokes. This is rather strange because ColumnsReview does detect these in other areas of the Explorer, but fails on the Desktop where only Control+NVDA+1 works to detect the item name.
I'm using NVDA Alpha 28879, ColumnsReview 5.3.20230812, and Windows 11 Version 22H2 (OS Build 22621.2134).
Thanks.

**Comment by ABuffEr** [comment]
Hi,
it's not so strange: in Windows 10, for example, Desktop list is different from folder lists.
Unfortunately I have no Windows 11 here... how did you set details view on Desktop? I'm not able to do it here.
If you can, going to list object by object navigation and report here the NVDA+F1 info could be useful.
Thanks.

**Comment by amirsol81** [comment]
@ABuffEr Thanks.
My View submenu on the Desktop has the following selected - or checked - items:
Large icons
Align icons to grid
Show desktop icons
However, unlike other folders in Explorer, the Desktop doesn't allow NVDA object navigation as attempting to move to the adjacent object moves me to a new Desktop item. Here's my NVDA+F1 log on an object in case you might find it useful:
INFO - globalCommands.script_navigatorObject_devInfo (15:28:59.702) - MainThread (4824):
Developer info for navigator object:
name: 'REAPER (x64)'
role: Role.LISTITEM
processID: 9552
roleText: None
states: State.SELECTABLE, State.FOCUSABLE, State.FOCUSED, State.SELECTED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_RecognizableFileObjectSysListView32EmittingDuplicateFocusEventsListItemIAccessible object at 0x0BBC8D70>
Python class mro: (<class 'NVDAObjects.Dynamic_RecognizableFileObjectSysListView32EmittingDuplicateFocusEventsListItemIAccessible'>, <class 'globalPlugins.nao.RecognizableFileObject'>, <class 'appModules.explorer.SysListView32EmittingDuplicateFocusEvents'>, <class 'NVDAObjects.IAccessible.sysListView32.ListItem'>, <class 'NVDAObjects.behaviors.RowWithFakeNavigation'>, <class 'NVDAObjects.behaviors.RowWithoutCellObjects'>, <class 'NVDAObjects.IAccessible.sysListView32.ListItemWithoutColumnSupport'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=1365, top=315, width=272, height=261)
value: None
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
appModule: AppModule(explorer, appName='explorer', processID=9552)
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.22621.1'
appModule.helperLocalBindingHandle: c_long(91981792)
windowHandle: 66012
windowClassName: 'SysListView32'
windowControlID: 1
windowStyle: 1442855488
extendedWindowStyle: 0
windowThreadID: 9556
windowText: 'FolderView'
displayText: 'REAPER (x64)'
IAccessibleObject: <POINTER(IAccessible) ptr=0x9a176e8 at bc450d0>
IAccessibleChildID: 15
IAccessible event parameters: windowHandle=66012, objectID=-4, childID=15
IAccessible accName: 'REAPER (x64)'
IAccessible accRole: ROLE_SYSTEM_LISTITEM
IAccessible accState: STATE_SYSTEM_SELECTED, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_MULTISELECTABLE, STATE_SYSTEM_VALID (19922950)
IAccessible accDescription: exception: (-2147352573, 'Member not found.', (None, None, None, 0, None))
IAccessible accValue: None

**Comment by amirsol81** [comment]
@ABuffEr Any progress on this with my NVDA log? Does it help in any way?

**Comment by ABuffEr** [comment]
Hi,
unfortunately not, because ColumnsReview intercepts the list, not listitems. I'll try to run Win11 in a VM tomorrow (downloaded the image yesterday).
Anyway, I noticed now you installed NAO add-on. There were conflicts with this add-on in the past, and actually I not use it. Can you try to temporarily disable to see what happens?
Thanks for your patience! :)

**Comment by amirsol81** [comment]
@ABuffEr Thanks for your efforts.
I disabled NAO, but it didn't affect my ColumnsReview issues both on the Win 11 Desktop and with the column headers in RSSOwlnix.

**Comment by ABuffEr** [comment]
Hi @amirsol81,
I have a Windows 11 PC now, but I cannot replicate your setting. I mean,
Large icons
Align icons to grid
Show desktop icons
are ok, but not sufficient to get details view (and it's not surprising, I think). Indeed, your situation was possible to Win10 too, but not at the moment (see [comments here](https://winaero.com/set-details-content-or-list-view-for-desktop-icons-in-windows-10/).
Additionally, I suspect that not even NVDA reads that columns, e.g. using ctrl+alt+arrows, right?

**Comment by amirsol81** [comment]
@ABuffEr Thanks for looking into this.
You're right - Control+NVDA+Arrows don't read the column headers on the Desktop. However, since JAWS retrieves them on the Desktop, I thought perhaps there should be a way to do it with ColumnsReview. I don't know how JAWS 2023 achieves that.

**Comment by amirsol81** [comment]
Sorry - I meant Control+ALT+arrows not Control+NVDA+arrows.

**Comment by ABuffEr** [comment]
I think that JAWS gets the UIA version of desktop FolderList, while NVDA continues - for now - to keep the IAccessible version. Note that I'm not really sure that JAWS does the right thing, or, in other words: columns are visually available? I think not.
Anyway, I'll wait NVDA changes on this, to avoid a useless work specifically for desktop list, and (very probably) to take advantages automatically whether this situation will supported in the future.
Let the issue open if you want.

**Comment by amirsol81** [comment]
@ABuffEr Thanks as always!
I'll keep it open to see what NVAccess does in the future.
